src/Stratis.Bitcoin/P2P/Protocol/Payloads/PayloadAttribute.cs

```diff
@@ -5,71 +5,99 @@
 
 namespace Stratis.Bitcoin.P2P.Protocol.Payloads
 {
+    /// <summary>
+    /// An attribute that enables mapping between command names and P2P netowrk types.
+    /// </summary>
     [AttributeUsage(AttributeTargets.Class)]
     public class PayloadAttribute : Attribute
     {
-        private static Dictionary<string, Type> nameToType;
-        private static Dictionary<Type, string> typeToName;
+        /// <summary>
+        /// The command name.
+        /// </summary>
         public string Name { get; set; }
 
-        static PayloadAttribute()
+        /// <summary>
+        /// Initialize a new instance of the object.
+        /// </summary>
+        /// <param name="commandName"></param>
+        public PayloadAttribute(string commandName)
         {
-            nameToType = new Dictionary<string, Type>();
-            typeToName = new Dictionary<Type, string>();
-
-            foreach (var pair in GetLoadableTypes(typeof(PayloadAttribute).GetTypeInfo().Assembly)
-                .Where(t => t.Namespace == typeof(PayloadAttribute).Namespace)
-                .Where(t => t.IsDefined(typeof(PayloadAttribute), true))
-                .Select(t =>
-                    new
-                    {
-                        Attr = t.GetCustomAttributes(typeof(PayloadAttribute), true).OfType<PayloadAttribute>().First(),
-                        Type = t
-                    }))
-            {
-                nameToType.Add(pair.Attr.Name, pair.Type.AsType());
-                typeToName.Add(pair.Type.AsType(), pair.Attr.Name);
-            }
+            this.Name = commandName;
         }
+    }
 
-        public PayloadAttribute(string commandName)
+    /// <summary>
+    /// A provider that maps <see cref="PayloadAttribute"/> types with <see cref="Message.Command"/>.
+    /// This is used by the P2P code to map and deserialize messages that are received from the tcp network to a concrete type.
+    /// </summary>
+    public class PayloadProvider
+    {
+        /// <summary>
+        /// A mapping between the payload type and the command name.
```

Comment:
Shouldn't this be `A mapping between the command name and the payload type`?

Reply:
ah yes thanks